src/marqo/core/document/document.py

```
@@ -137,7 +137,8 @@ def partial_update_documents(self, partial_documents: List[Dict], marqo_index) \
                 VESPA_FIELD_ID, INT_FIELDS, FLOAT_FIELDS, VESPA_DOC_FIELD_TYPES, VESPA_DOC_CREATE_TIMESTAMP], schema = marqo_index.schema_name)
             responses = get_batch_response.responses
             for resp in responses:
-                existing_vespa_documents[resp.document.fields[VESPA_FIELD_ID]] = resp.document.dict()
+                if resp.document:
```

Why did we add this line? Also note these checks tend to be risky as it evals to false if `resp.document` is None or empty list/dict. Can we do a more specific condition if it's required?

---

I've added it because when debugging I noticed that there was a case when resp.document was None so resp.document.dict() throwing the error `[error 'NoneType' object has no attribute '__dict__']`

---

Let's track in this thread. Do we know why it returns None? should we raise an error in this case instead?

---

It returns none when a user sends a update request for a doc that doesn't exist but the update request contains a map. 
Since it contains a map we will try to get docs which contains maps but since the doc doesn't exist in the index resp.document will be none. 

I think we can raise an error in this case (similar to how we handle other errors i.e by adding to unsuccessful_docs & sending it as response).

---

yes, if a user tries to update an non-existing doc, I think we should return 404 early, instead of updating Vespa and expect a 412.

---

to maintain consistency we will throw a 400